### ISSUE TITLE
Cancel close doesn't cancel

### ISSUE THREAD

**Issue by rappen** [MscrmTools/XrmToolBox#805]
Example:
Scene: After you have used FetchXMLBuilder and click close you get prompted to Save.
Bug: If you click Cancel, it is treated the same as clicking No.

Ref https://github.com/rappen/FetchXMLBuilder/issues/218

**Comment by murthysrvn** [comment]
This is happening because the plugin in creating a setting file Rappen.XTB.FXB in AppData Settings which has the FetchXML is not cleared on close\cancel. Hope this helps in fixing the bug.

**Comment by rappen** [comment]
Researched a bit... and the ball bounces back, with this scenario:
https://www.screencast.com/t/UoD0GoUSfGBb

I have also made the behavior in FXB more robust to reduce impact from the tool.

**Comment by MscrmTools** [comment]
Still an issue?

**Comment by rappen** [comment]
Yes. But actually slightly different issue. Which might be worse, since me cancelling the close is ignored...
https://www.screencast.com/t/OWgVDQbUB

**Comment by murthysrvn** [comment]
@rappen,

I tried reproduce this but i can't not sure if the fix is already in place. I still see that without saving if i close the tool the fetch xml is being restored. However i noticed that it's a new file created per connection Rappen.XTB.FXB_connectionName.xml. I assume if i create a new connection i will be seeing the empty fetch xml again.

**Comment by rappen** [comment]
@murthysrvn yes the persistence of the queries is by conneciton.

**Comment by BetimBeja** [comment]
Hello @MscrmTools and @rappen 
https://user-images.githubusercontent.com/11160171/130398688-b01958c7-2f80-4798-a0d3-ed3d970045ce.mp4

I replicated this bug with the latest version of XrmToolBox. As you can see from the video I think the problem is with XrmToolBox when closing all tools, it doesn't consider the specific tools response.
